### PR TITLE
Rebase x86

### PR DESCRIPTION
**Pull request by nicoTolly**
Move x86 sources into its own backend crate

### CODE REVIEW

File: src/codegen/variable.rs [hunk 1]
```diff
@@ -207,58 +206,4 @@ mod tests {
     {
         content.iter().cloned().collect()
     }
-
-    /// Ensures chained aliases work correctly.
-    #[test]
```
Comment (ulysseB) [inline src/codegen/variable.rs]: Why is this test removed ?

File: src/device/mod.rs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 //! Code generation and candidate evaluation for specific targets.
 pub mod fake;
-pub mod x86;
+#[cfg(feature = "mppa")]
+pub mod mppa;
```
Comment (ulysseB) [inline src/device/mod.rs]: Why are you adding back mppa as a module here ?

File: src/codegen/variable.rs (hunk 1 above)
Comment (nicoTolly) [reply src/codegen/variable.rs]: It is the circular dependency we talked about

File: src/device/mod.rs (hunk 2 above)
Comment (nicoTolly) [reply src/device/mod.rs]: Merge error

File: src/codegen/variable.rs (hunk 1 above)
Comment (ulysseB) [reply src/codegen/variable.rs]: Ok. Fine for me.

**Comment by nicoTolly** [comment]
bors retry

**Comment by nicoTolly** [comment]
bors retry

**Comment by nicoTolly** [comment]
bors retry

**Comment by nicoTolly** [comment]
bors retry

**Comment by nicoTolly** [comment]
bors retry

**Pull request merged**